akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -233,6 +233,24 @@ private[http] object StreamUtils {
       def open(): Unit = promise.success(())
     }
   }
+
+  /**
+   * Returns a flow that is almost identity but doesn't pass cancellation from downstream to upstream.
+   *
+   * Note: This might create a resource leak if the upstream never completes. External measures
+   * need to be taken to ensure that the upstream will always complete.
+   */
+  def ignoreCancellation[T]: Flow[T, T, NotUsed] = Flow.fromGraph(new IgnoreCancellation)
+  class IgnoreCancellation[T] extends SimpleLinearGraphStage[T] {
```

Comment:
I guess the comment is not quite accurate. Cancellation is not ignored in the sense that the stage will sent additional elements after it received cancellation (that is already prevented by the GraphStage). It is just not propagated to the upstream side of this flow in this case. (The default behavior when receiving cancellation will call `completeStage` which is correct for most flow-like stages but for many others it isn't.)

Also, it's only a utility class. As explained (also in the other comment in Http.scala) it is only used in circumstances where the overall setup ensures that a flow (or complete graph setup) is closed from the other side.

And, yes, `final` would make sense, it already lives in a `private[http]` object.